method-README-template.md

```diff
@@ -1,65 +1,71 @@
+6. It should also provide different configuration parameters to assist in altering the behavior of the method for a related use case. 
 
-## Keywords
-Few keywords placed after description highlighting the nature of the method 
 
-## Science Usecase(s)
-- Include usecases from social sciences that would make this method applicable in a certain scenario. 
-The use cases or research questions mentioned should arise from the latest social science literature cited in the description.
-For example,
-How to collect 4chan data from on political discourse from dates ..
- 
+## Keywords
+- Keyword 1 (3 to 5 keywords highlighting the purpose of method)
+- Keyword 2
+- Keyword 3   
 
-## Repo Structure
-- Explain the overall structure of the method, including directories, key files, and their functions.
-For example,
-The tool's architecture includes a src/ directory for core scripts, with requester.py handling data collection, board.py managing board-specific requests, and utils.py for auxiliary functions. Data is stored in a data/ directory created upon initiation, and documentation is available in docs/.
+## Social Science Use Case(s)
+1. Include use case(s) or research question(s) from social sciences that would make this method applicable in a certain scenario.
+2. They should arise from the latest social science literature cited in the description. 
 
+## Git Repository Structure
+1. List files and their purpose as a tree-like structure.
+2. Explain the script files and their purpose.
+3. Explain the input, output and other intermediate files produced.
+4. Explain the configuration and other utility files.
 
 ## Environment Setup
-- Setting up the envornment to run this method locally
-- Installing all the packages and libraries with specific versions required to run this method
-For example, 
-Requires Python 3.10.2 or 3.11.4. Suitable for environments focused on data collection and analysis.
-Dependencies are listed in requirements.txt and can be installed via pip install -r requirements.txt to ensure the tool functions correctly.
-
-
-## Hardware Requirements (Optional)
-- The hardware requirements may be needed in specific cases when a method is known to require more memory/compute power. 
-- The method need to be executed on a specific architecture (GPUs, Hadoop cluster etc.)
+1. Provide configuration files preserving dependencies i.e., `environment.yml`, `requirements.txt`, `install.R` (including specific versions).
+2. Provide steps/commands to install all dependencies required to run this method.
+3. Load default configurations to execute the method without needing user envolvement e.g., for input or paths to resources. Store the configuration files separately (e.g., in config.json) where the users can change the parameters to alter the behavior of the method without directly interacting with the code.
```

Comment:
Recommend a separately configuration file increases the required technical knowledge from authors. Also, this is incompatible with the design partner implemented by Python and R packages.

My recommendation is to completely drop this 3rd item.

Reply:
Actually, I totally don't understand the need for this `config.json`, let's say for R/Python packages. I also recommend drop this.